src/lib.rs

```diff
@@ -188,6 +188,24 @@ use core::slice;
 use core::str;
 use core_alloc::alloc::{alloc, dealloc, Layout};
 
+/// An error returned from [`Bump::try_alloc_try_with`].
+#[derive(Clone, PartialEq, Eq, Debug)]
+pub enum AllocOrInitError<E> {
```

Comment:
I went with the slightly more general `AllocOrInitError` over `TryAllocTryWith` here, after all.
I hope exposing the variants is alright, but it's still a purpose-bound error so I *think* this is okay.

The type should also implement `Error` in my opinion, but that would touch on AllocError too, so I'll wait with that for now.

Reply:
>  I hope exposing the variants is alright, but it's still a purpose-bound error so I _think_ this is okay.

Yeah this is fine.

> The type should also implement `Error` in my opinion, but that would touch on AllocError too, so I'll wait with that for now.

This crate isn't `std` so we can't implement the `Error` trait, as it isn't available in `core`.